packages/integrations/components/pico/observer.js

```diff
@@ -0,0 +1,200 @@
+        setCanComment(true);
+      }
+      // If the status is registered, set the isRegistered flag to true.
+      if ('registered' === status) {
+        setIsRegistered(true);
+      }
+      // If the status is paying, set the isPaying flag to true.
+      if ('paying' === status) {
+        setIsPaying(true);
+      }
+    }
+
+    // If the user was previously recognized as registered or paying, but
+    // is then anonymized, update the state values accordingly.
+    if ((isRegistered || isPaying) && 'anonymous' === status) {
+      if (isRegistered) {
+        setIsRegistered(false);
+      }
+
+      if (isPaying) {
+        setIsPaying(false);
+      }
+
+      // Clear the user's email address.
+      setEmail(undefined);
+      // Set the user's status to anonymous.
+      setIsAnonymous(true);
+    }
+  };
+
+  const observerCallback = debounce((mutations) => {
+    mutations.forEach((mutation) => {
+      const { target, type, attributeName } = mutation;
+
+      if ('attributes' === type && 'data-pico-status' === attributeName) {
+        manageLocalState(target);
+      }
+    });
+  }, 250);
+
+  // Declare a ref for the `useMutationObserver` hook to watch.
+  const signalRef = useRef();
+  // Mount the MutationObserver.
+  useMutationObserver(signalRef, observerCallback);
+
+  // Define the global dispatch function.
+  const dispatch = useDispatch();
+
+  // Define a function to summon an upgrade prompt for Coral SSO.
+  const showUpgradeModal = useCallback(
+    () => dispatch(actionRequireUpgrade()),
+    [dispatch]
+  );
+  // Define a function to dispatch Pico user verification requests.
+  const sendVerificationRequest = useCallback(
+    (user) => dispatch(actionVerifyPicoUser(user)),
+    [dispatch]
+  );
+  // Define a function to dispatch a plan upgrade requirement message.
+  const receivePlanUpgrade = useCallback(
+    () => dispatch(actionReceivePicoPlanUpgrade()),
+    [dispatch]
+  );
+  // Define a function to dispatch logout requests. This function is explicitly
+  // passed through props because the observer may want to cause side-effects
+  // in the application outside of the context of Pico.
+  const requestLogout = useCallback(
+    () => dispatch(logoutRequestAction()),
+    [dispatch]
+  );
+
+  // Grab the status of the upgrade modal's visibility from the Redux store.
+  const upgradeModalVisible = useSelector(showUpgradeModalSelector);
+  const upgradeModalDismissed = useSelector(upgradeModalDismissedSelector);
+
+  useEffect(() => {
+    // If the user has been set to anonymous, dispatch the logout request.
+    if (canComment && isAnonymous) {
+      requestLogout();
+      setCanComment(false);
+      // Exit the effect early.
+      return;
+    }
+
+    if (! accessToken) {
+      // If the user is registered but non-paying and the upgrade modal
+      // has not been show, update the store to trigger the modal.
+      if (
+        (
+          isRegistered ||
+          isPaying
```

Comment:
If I'm reading your inline comment correctly, should this be `! isPaying`?

Reply:
That comment was incorrect. Updating 😄